conformance_tests/core/test_cmdqueue/src/test_cmdqueue.cpp

```diff
@@ -235,6 +235,155 @@ TEST_P(
+        char_input[j] = verification_buffer[j];
+      }
+
+      host_buffers.push_back(static_cast<uint8_t *>(host_buffer));
+
+      void *deive_buffer = nullptr;
+      ze_device_mem_alloc_desc_t deviceDesc = {};
+      deviceDesc.stype = ZE_STRUCTURE_TYPE_DEVICE_MEM_ALLOC_DESC;
+      deviceDesc.ordinal = 0;
+      deviceDesc.flags = 0;
+      deviceDesc.pNext = nullptr;
+
+      EXPECT_EQ(ZE_RESULT_SUCCESS,
+                zeMemAllocDevice(context, &deviceDesc, buff_size_bytes,
+                                 buff_size_bytes, device, &deive_buffer));
+      device_buffers.push_back(static_cast<uint8_t *>(deive_buffer));
+
+      ze_command_list_handle_t command_list_regular;
+      cmdListDesc.commandQueueGroupOrdinal =
+          lzt::getComputeQueueGroupOrdinal(device);
+      cmdListDesc.flags = 0;
+      EXPECT_EQ(ZE_RESULT_SUCCESS,
+                zeCommandListCreate(context, device, &cmdListDesc,
+                                    &command_list_regular));
+      EXPECT_NE(nullptr, command_list_regular);
+
+      // Copy from host-allocated to device-allocated memory
+      EXPECT_EQ(ZE_RESULT_SUCCESS,
+                zeCommandListAppendMemoryCopy(
+                    command_list_regular, deive_buffer, host_buffer,
+                    buff_size_bytes, nullptr, 0, nullptr));
+
+      EXPECT_EQ(ZE_RESULT_SUCCESS,
+                zeCommandListAppendBarrier(command_list_regular, nullptr, 0,
+                                           nullptr));
+
+      // Copy from device-allocated memory back to host-allocated memory
+      EXPECT_EQ(ZE_RESULT_SUCCESS,
+                zeCommandListAppendMemoryCopy(command_list_regular, host_buffer,
+                                              deive_buffer, buff_size_bytes,
+                                              nullptr, 0, nullptr));
+
+      EXPECT_EQ(ZE_RESULT_SUCCESS, zeCommandListClose(command_list_regular));
+      list_of_command_lists.push_back(command_list_regular);
+    }
+  }
+
+  void TearDown() override {
+    for (uint32_t i = 0; i < params.num_command_lists; i++) {
+
+      EXPECT_EQ(
+          0, memcmp(host_buffers.at(i), verification_buffer, buff_size_bytes));
+
+      EXPECT_EQ(ZE_RESULT_SUCCESS,
+                zeCommandListDestroy(list_of_command_lists.at(i)));
+      lzt::free_memory(host_buffers.at(i));
+      lzt::free_memory(device_buffers.at(i));
+    }
+    lzt::destroy_command_queue(command_queue);
+  }
+
+  static const uint32_t buff_size_bytes = 12;
+  uint8_t verification_buffer[buff_size_bytes];
+  CustomExecuteParams params = GetParam();
+
+  ze_command_queue_handle_t command_queue = nullptr;
+  std::vector<ze_command_list_handle_t> list_of_command_lists;
+
+  std::vector<uint8_t *> host_buffers;
+  std::vector<uint8_t *> device_buffers;
+
+  ze_command_list_handle_t command_list_immediate;
+};
+
+class zeCommandListImmediateAppendCommandListsExpTestsHostSynchronize
+    : public zeImmediateCommandListAppendCommandListsExpTests {};
```

Comment:
Why child class with no additional definitions here? Just to name them specific for host synchronize? Do we use zeImmediateCommandListAppendCommandListsExpTests somewhere else?

Reply:
Good question. I kept it this way as I will be putting in new PRs for new ULTs so might need to extend the base class.